lib/saluki-components/src/sources/checks/builder/python/python_modules.rs

```diff
@@ -120,3 +274,137 @@ pub mod datadog_agent {
+            match event {
+                Event::Metric(_) => metric_count += 1,
+                Event::ServiceCheck(sc) => {
+                    service_check_count += 1;
+                    assert_eq!(sc.name(), "test.service_check");
+                    assert_eq!(sc.status(), CheckStatus::Ok);
+                    assert_eq!(sc.message().unwrap(), "All systems operational");
+                }
+                Event::EventD(ev) => {
+                    event_count += 1;
+
+                    if ev.title() == "Test Event Title" {
+                        // Full event test
+                        assert_eq!(ev.text(), "This is a test event message from Python");
+                        assert_eq!(ev.hostname().unwrap(), "test-event-hostname");
+                        assert_eq!(ev.alert_type().unwrap(), AlertType::Warning);
+                        assert_eq!(ev.priority().unwrap(), Priority::Low);
+                        assert_eq!(ev.aggregation_key().unwrap(), "test-aggregation-key");
+                        assert_eq!(ev.source_type_name().unwrap(), "python_test");
+                        assert_eq!(ev.timestamp().unwrap(), 1234567890);
+
+                        let tags = ev.tags();
+                        assert!(tags.into_iter().any(|tag| tag.as_str() == "event:test"));
+                        assert!(tags.into_iter().any(|tag| tag.as_str() == "source:python"));
+                        assert!(tags.into_iter().any(|tag| tag.as_str() == "priority:high"));
+                    } else if ev.title() == "Minimal Event" {
+                        // Minimal event test
+                        assert_eq!(ev.text(), "This event has only required fields");
+                        assert!(ev.hostname().is_none());
+                        assert!(ev.aggregation_key().is_none());
+                        assert!(ev.source_type_name().is_none());
+                        assert!(ev.timestamp().is_none());
+                        assert!(ev.tags().into_iter().count() == 0);
+                        // Default values should be set for priority and alert_type
+                        assert_eq!(ev.priority().unwrap(), Priority::Normal);
+                        assert_eq!(ev.alert_type().unwrap(), AlertType::Info);
+                    } else {
+                        panic!("Unexpected event title: {}", ev.title());
+                    }
+                }
+            }
+        }
+
+        assert!(metric_count >= 2, "Expected at least 2 metrics, got {}", metric_count);
+        assert_eq!(
+            service_check_count, 1,
+            "Expected 1 service check, got {}",
+            service_check_count
+        );
+        assert_eq!(event_count, 2, "Expected 2 events, got {}", event_count);
+    }
+
+    struct TestResults {
+        hostname: String,
+        config_value: String,
+    }
+
+    #[test]
+    fn test_python_datadog_agent_integration() {
+        std::env::set_var("DD_TEST_FOO", "bar");
```

Comment:
This is very brittle -- setting environment variables at runtime -- but there's no better option at the moment for what you're trying to do... so I'm just saying this to remind myself that we need a answer for mocking out stuff in configuration.